sympy/polys/polytools.py

```diff
@@ -5949,6 +5947,9 @@ def _symbolic_factor_list(expr, opt, method):
                         other.append((f, k))
 
                 factors.append((_factors_product(other), exp))
+    if method == 'sqf':
+        factors = [(reduce(mul, (f for f, _ in factors if _ == k)), k)
+                   for k in set(dict(factors).values())]
```

Comment:
It could be possible to avoid construction of a dictionary. Maybe something like this:

     set(i for _, i in factors)